lib/lutf8/lutf8.c

```diff
@@ -50,11 +46,11 @@ static lua_Integer u_posrelat (lua_Integer pos, size_t len) {
 ** entry forces an error for non-ascii bytes with no continuation
 ** bytes (count == 0).
 */
-static const char *utf8_decode (const char *s, l_uint32 *val, int strict) {
-  static const l_uint32 limits[] =
-        {~(l_uint32)0, 0x80, 0x800, 0x10000u, 0x200000u, 0x4000000u};
+static const char *utf8_decode (const char *s, uint32_t *val, int strict) {
```

Comment:
what about just defining `lua_Unsigned` and `l_uint32` for this file? the fewer the edits the better